src/handlers/gestures/gestureBuilder.ts

```diff
@@ -0,0 +1,126 @@
+import { SimpleGesture } from './simpleGestures';
+
+enum Relation {
+  Simultaneous,
+  Exclusive,
+  After,
+  RequireToFail,
+}
+
+type PendingGesture = {
+  relation: Relation;
+  gesture: SimpleGesture;
+};
+
+export class GestureBuilder {
+  private log: Array<PendingGesture> = [];
+
+  constructor(base: SimpleGesture) {
+    this.log.push({ relation: Relation.Exclusive, gesture: base });
+  }
+
+  simultaneousWith(gesture: SimpleGesture): GestureBuilder {
+    this.log.push({ relation: Relation.Simultaneous, gesture: gesture });
+
+    return this;
+  }
+
+  exclusiveWith(gesture: SimpleGesture): GestureBuilder {
+    this.log.push({ relation: Relation.Exclusive, gesture: gesture });
+
+    return this;
+  }
+
+  after(gesture: SimpleGesture): GestureBuilder {
+    this.log.push({ relation: Relation.After, gesture: gesture });
+
+    return this;
+  }
+
+  requireToFail(gesture: SimpleGesture): GestureBuilder {
+    this.log.push({ relation: Relation.RequireToFail, gesture: gesture });
+
+    return this;
+  }
+
+  build(): BuiltGesture {
+    let result = new BuiltGesture(this.prepare);
+
+    result.gestures = [];
+
+    for (const pg of this.log) {
+      result.gestures.push(pg.gesture);
+    }
+
+    return result;
+  }
+
+  prepare = () => {
+    let simultaneous = [];
+    let after = [];
+    let waitFor = [];
```

Comment:
```suggestion
    const simultaneous = [];
    const after = [];
    const waitFor = [];
```

Reply:
There are more variables with `let` that can be `const` but I won't add comments to them to decrease the signal to noise ratio 😄

Reply:
Fixed in fdabf483d5407f3ea1c8fb6f0c75309a533efe41